src/inspect_ai/tool/_tools/_computer/_resources/tool/_x11_client.py

```diff
@@ -105,138 +92,209 @@ def __init__(self):
 
         self.xdotool = f"{self._display_prefix}xdotool"
 
-    async def __call__(
```

Comment:
The big design change here is to push all of the magical, untyped shenanigans up to the caller. This class now has nothing but discrete methods to do all of the things, and no `Any`'s anymore.